src/NServiceBus.Persistence.AzureStorage/Subscriptions/AzureSubscriptionStorage.cs

```diff
@@ -142,5 +190,11 @@ public int GetHashCode(Subscriber obj)
                 return (obj.Endpoint ?? string.Empty).Length;
             }
         }
+
+        internal class CacheItem
+        {
+            public DateTime Stored;
+            public Task<IEnumerable<Subscriber>> Subscribers;
```

Comment:
Personally, I find the last comment to add more complexity than clarity.
Get a task, cache it. Await for the result. Next time cached task is awaited, it's going to return the straight result.